build/parseChangelog.c

```diff
@@ -178,13 +179,17 @@ static int dateToTimet(const char * datestr, time_t * secs, int * date_words)
 
     if (*secs == -1) goto exit;
 
+    if (tzname[1][0] == 0)
+	rpmlog(RPMLOG_WARNING, _("time zone in %%changelog not in tz database: %s\n"), datestr);
```

Comment:
Um. Why not just "unknown timezone in changelog: %s"? Speaking about database sounds like its a misconfiguration or something like that.

Reply:
In the new version I changed the current warning, to be close to yours. But it still give an additional information that the time zone ID must be contained in the tz database (thus it can not be e.g. time zone abbreviation like CEST). So it additionally helps to indicate how to correct the problem.

Reply:
But it's not like *you're* consulting a timezone database here. What tzname contains is up to the libc implementation, https://pubs.opengroup.org/onlinepubs/9699919799/functions/tzset.html or https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html#tag_08 say nothing at all about any databases, much less that tzname[1] contains a value from the tz database.

 In fact, tzname[1] is dst, of which the standard says:
> std and dst
    Indicate no less than three, nor more than {TZNAME_MAX}, bytes that are the designation for the standard (std) or the alternative (dst -such as Daylight Savings Time) timezone. Only std is required; if dst is missing, then the alternative time does not apply in this locale. 

This is far from my area of expertise but based on the above, I don't see how you can determine the validity from tzname[1].
Further, tzname is a string so for an empty string you should be comparing for character '\0', not 0 the number. It might not make a difference to the computer but these things matter to the reader.